src/bin/ayb.rs

```diff
@@ -360,26 +374,58 @@ async fn main() -> std::io::Result<()> {
                 }
             }
         } else if let Some(matches) = matches.subcommand_matches("query") {
-            if let (Some(entity_database), Some(query), Some(format)) = (
+            if let (Some(entity_database), Some(format)) = (
                 matches.get_one::<EntityDatabasePath>("database"),
-                matches.get_one::<String>("query"),
                 matches.get_one::<OutputFormat>("format"),
             ) {
-                match client
+                if let Some(query) = matches.get_one::<String>("query") {
+                    match client
                     .query(&entity_database.entity, &entity_database.database, query)
                     .await
-                {
-                    Ok(query_result) => {
-                        if !query_result.rows.is_empty() {
-                            match format {
-                                OutputFormat::Table => query_result.generate_table()?,
-                                OutputFormat::Csv => query_result.generate_csv()?,
+                    {
+                        Ok(query_result) => {
+                            display_query_result(&query_result, format)?;
+                        }
+                        Err(err) => {
+                            println!("Error: {}", err);
+                        }
+                    }   
+                } else {
+                    println!("Launching an interactive session for {}/{}", entity_database.entity, entity_database.database);
+                    
+                    match DefaultEditor::new() {
+                        Ok(mut rl) => loop {
+                            let line = rl.readline(">>> ");
+                            match line {
+                                Ok(line) if line.is_empty() => {}
+                                Ok(query) => {
+                                    let result = rl.add_history_entry(query.as_str());
```

Comment:
I'm similarly OK using ? or not checking the result here, since we're not saving the history to a file.

Reply:
We could add an `.unwrap()` there to avoid checking the result, it's not necessary to check imo.

I don't know how can we add a `?` there because seems that there is a mismatch on the different error types :(

Reply:
Ugh error types. :( I'll see if there's a trick I can play with it, or if it's not worth it and perhaps an unwrap in the client is fine.